tasks/build.yml

```diff
@@ -56,3 +56,18 @@
     insertafter: EOF
     line: "export PATH=/home/{{ nodejs.workspace_user }}/.npm-global/bin:$PATH"
   when: nodejs.workspace_user
+
+- name: Install NPMs
+  become: yes
+  npm:
+    executable: "{{ item.executable | default('') }}"
+    global: "{{ item.global | default('no' if item.path is defined else 'yes') }}"
+    ignore_scripts: "{{ item.ignore_scripts | default('no') }}"
+    name: "{{ item.name }}"
+    path: "{{ item.path | default('') }}"
+    production: "{{ item.production | default('no') }}"
+    registry: "{{ item.registry | default('') }}"
+    state: "{{ item.state | default('present') }}"
+    version: "{{ item.version | default('') }}"
+  with_items: "{{ nodejs.npms }}"
+  when: nodejs.npms
```

Comment:
The preference is to define all possible variables in defaults.yml so it's easy to see all available variables for any role, with this is in mind can you define npms in defaults.yml like so:

```YAML
---

nodejs:
  npms: [ ]
  # 4 or 5
  version: 4
  workspace_user: ~
```
After this change you will not need the when clause, if nodejs.npms is empty Ansible will skip the task automatically.

Reply:
Thank you!  I originally did not have all those `| default()` clauses, passing all variables through instead, but:

> [DEPRECATION WARNING]: Using variables for task params is unsafe, especially if
 the variables come from an external source like facts. This feature will be
removed in version 2.6.

I have changed the code as per your comment.